shardingsphere-underlying/shardingsphere-common/src/test/java/org/apache/shardingsphere/underlying/common/config/DataSourceConfigurationTest.java

```diff
@@ -83,4 +88,25 @@ public void assertAddAlias() {
         assertThat(actual.getProperties().get("jdbcUrl"), is(actual.getProperties().get("url")));
         assertThat(actual.getProperties().get("username"), is(actual.getProperties().get("user")));
     }
+    
+    @Test
+    public void assertGetDataSourceConfigurationWithConnectionInitSqls() throws SQLException {
```

Comment:
Here is a integration test.
I suggestion:
1. if do this way, it's neccesary to use HikariDataSource instead of BasicDatasource.
2. creat a table in h2 db, and set an insert sql to initSql, then check it work well by a select and compare.

Or anothor approach: 
just mock a jdbc driver behind HikariDataSource, check hikariDataSource send the initSql to mock jdbc connection/statement.

Reply:
connectionInitSqls in HikariDataSource is a String type, but we want to verify a collection type for this issue. also I found that collection type property not existed in in HikariDataSource, it seems that DBCP2 is ok for that

Reply:
> connectionInitSqls in HikariDataSource is a String type, but we want to verify a collection type for this issue. also I found that collection type property not existed in in HikariDataSource, it seems that DBCP2 is ok for that

@kimmking @cherrylzhao In HikariDataSource, the configuratoin is connectionInitSql, and it is a String type, i also veryfied HikariDataSource , it's ok. DBCP2 and Druid support Connection type,  the only difference is the return parameter of `getConnectionInitsqls` method, DBCP2 return List type, Druid return Connection type, I have done compatibility processing in the pr.